resilience4j-spring/src/main/java/io/github/resilience4j/circuitbreaker/configure/RxJava2CircuitBreakerAspect.java

```diff
@@ -0,0 +1,115 @@
+import io.reactivex.MaybeSource;
+import io.reactivex.Observable;
+import io.reactivex.ObservableSource;
+import io.reactivex.Single;
+import io.reactivex.SingleSource;
+
+/**
+ * the Rx circuit breaker logic support for the spring AOP
+ * conditional on the presence of Rx classes on the spring class loader
+ */
+
+@Component
+@Conditional(value = {InjectRxJava2Aspect.class})
+class RxJava2CircuitBreakerAspect implements CircuitBreakerAspectExt {
+
+	private static final Logger logger = LoggerFactory.getLogger(CircuitBreakerAspect.class);
+	private final Set<Class> rxSupportedTypes = newHashSet(ObservableSource.class, SingleSource.class, CompletableSource.class, MaybeSource.class, Flowable.class);
+
+	public RxJava2CircuitBreakerAspect() {
+	}
+
+	@SafeVarargs
+	private static <T> Set<T> newHashSet(T... objs) {
+		Set<T> set = new HashSet<>();
+		Collections.addAll(set, objs);
+		return Collections.unmodifiableSet(set);
+	}
+
+	/**
+	 * @param returnType the AOP method return type class
+	 * @return boolean if the method has Rx java 2 rerun type
+	 */
+	@SuppressWarnings("unchecked")
+	@Override
+	public boolean matchReturnType(Class returnType) {
+		return rxSupportedTypes.stream().anyMatch(classType -> classType.isAssignableFrom(returnType));
+	}
+
+	/**
+	 * @param proceedingJoinPoint Spring AOP proceedingJoinPoint
+	 * @param circuitBreaker      the configured circuitBreaker
+	 * @param methodName          the method name
+	 * @return the result object
+	 * @throws Throwable exception in case of faulty flow
+	 */
+	@SuppressWarnings("unchecked")
+	@Override
+	public Object handle(ProceedingJoinPoint proceedingJoinPoint, CircuitBreaker circuitBreaker, String methodName) throws Throwable {
+		CircuitBreakerUtils.isCallPermitted(circuitBreaker);
+		CircuitBreakerOperator circuitBreakerOperator = CircuitBreakerOperator.of(circuitBreaker);
+		long start = System.nanoTime();
+		try {
+			Object returnValue = proceedingJoinPoint.proceed();
+			if (returnValue instanceof ObservableSource) {
+				Observable observable = (Observable) returnValue;
+				return observable.lift(circuitBreakerOperator);
+			} else if (returnValue instanceof SingleSource) {
+				Single single = (Single) returnValue;
+				return single.lift(circuitBreakerOperator);
+			} else if (returnValue instanceof CompletableSource) {
+				Completable completable = (Completable) returnValue;
+				return completable.lift(circuitBreakerOperator);
+			} else if (returnValue instanceof MaybeSource) {
+				Maybe maybe = (Maybe) returnValue;
+				return maybe.lift(circuitBreakerOperator);
+			} else if (returnValue instanceof Flowable) {
+				Flowable flowable = (Flowable) returnValue;
+				return flowable.lift(circuitBreakerOperator);
+			} else {
+				throw new IllegalArgumentException("Not Supported type for the circuit breaker in web flux :" + returnValue.getClass().getName());
+			}
+		} catch (Throwable throwable) {
+			long durationInNanos = System.nanoTime() - start;
```

Comment:
I think we can remove it. We don't have to hande the throwable. This exception signals a development bug and wrong usage of RxJava2. Must not be captured by a CircuitBreaker.